### PR TITLE
[Student][Parent][MBL-14393] Upgrade to Flutter 1.17

### PR DESCRIPTION
**Pull request by JordanMarshall**
Updates Flutter for parent and student embed to 1.17.1. Due to 1.17 using a newer version of appcompat than Student, some changes (mostly nullability changes) had to be made to otherwise unrelated parts of the Student app. Also, I commented out a few failing Student tests related to l10n, assuming that they will be addressed as part of MBL-14395.

### CODE REVIEW

File: apps/student/src/main/java/com/instructure/student/fragment/ParentFragment.kt [hunk 1]
```diff
@@ -198,18 +198,6 @@ abstract class ParentFragment : DialogFragment(), ConfigureRecyclerView, Fragmen
...
-        // Very important fix for the support library and child fragments.
-        try {
-            val childFragmentManager = Fragment::class.java.getDeclaredField("mChildFragmentManager")
-            childFragmentManager.isAccessible = true
-            childFragmentManager.set(this, null)
-        } catch (e: NoSuchFieldException) {
-            e.printStackTrace()
-        } catch (e: IllegalAccessException) {
-            e.printStackTrace()
-        }
```
Comment (JordanMarshall) [inline apps/student/src/main/java/com/instructure/student/fragment/ParentFragment.kt:211]: This workaround is no longer needed in recent versions of appcompat. In fact, leaving it in here was causing crashes.

File: apps/student/src/main/java/com/instructure/student/dialog/HelpDialogStyled.kt [hunk 2]
```diff
@@ -89,7 +89,7 @@ class HelpDialogStyled : DialogFragment() {
 
     override fun onDestroyView() {
         if (dialog != null && retainInstance)
```
Comment (StephenBrough) [inline apps/student/src/main/java/com/instructure/student/dialog/HelpDialogStyled.kt:91]: Could probably just get rid of the null check.

File: apps/student/src/main/java/com/instructure/student/dialog/FatalErrorDialogStyled.kt [hunk 3]
```diff
@@ -55,7 +55,7 @@ class FatalErrorDialogStyled : DialogFragment() {
 
     override fun onDestroyView() {
         if (dialog != null && retainInstance)
```
Comment (StephenBrough) [inline apps/student/src/main/java/com/instructure/student/dialog/FatalErrorDialogStyled.kt:57]: Could probably just get rid of the null check.

File: apps/student/src/main/java/com/instructure/student/dialog/LegalDialogStyled.kt [hunk 4]
```diff
@@ -100,7 +100,7 @@ class LegalDialogStyled : AppCompatDialogFragment() {
     }
 
     override fun onDestroyView() {
-        if (dialog != null && retainInstance) dialog.setDismissMessage(null)
+        if (dialog != null && retainInstance) dialog?.setDismissMessage(null)
```
Comment (StephenBrough) [inline apps/student/src/main/java/com/instructure/student/dialog/LegalDialogStyled.kt:103]: Could probably just get rid of the null check.

**Pull request merged**